sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/JsonSerializable.java

```diff
@@ -154,6 +154,15 @@ public void populatePropertyBag() {
         return getMapper().convertValue(this.propertyBag, HashMap.class);
     }
 
+    @SuppressWarnings("unchecked")
+    public <T> Map<String, T> getMap(String propertyKey) {
+        if (this.propertyBag.has(propertyKey)) {
+            Object value = this.get(propertyKey);
+            return (Map<String, T>) getMapper().convertValue(value, HashMap.class);
```

Comment:
Error check  is value is not map ?

Reply:
Done by the convertValue. It throws IllegalArgumentException, and thats what we would throw even if we check.